mkosi/__init__.py

```diff
@@ -3799,13 +3536,26 @@ def ensure_networkd(config: MkosiConfig) -> bool:
     return True
 
 
+def nspawn_knows_arg(arg: str) -> bool:
+    # Specify some extra incompatible options so nspawn doesn't try to boot a container in the current
+    # directory if it has a compatible layout.
+    return "unrecognized option" not in run(["systemd-nspawn", arg,
+                                            "--directory", "/dev/null", "--image", "/dev/null"],
+                                            stdout=subprocess.DEVNULL, stderr=subprocess.PIPE, check=False,
+                                            text=True).stderr
+
+
 def run_shell(config: MkosiConfig) -> None:
```

Comment:
Any reason not to move `run_shell` into `run.py` as well?

Reply:
Depends on `MkosiConfig` and we can't import `backend` into `run`. Doing this requires splitting things up more, so I'm not doing it for now

Reply:
PR is big already, let's leave moving the other run() functions in for another PR